include/unifex/timed_single_thread_context.hpp

```diff
@@ -51,163 +48,172 @@ class timed_single_thread_context {
+    friend scheduler;
 
-       private:
-        template <typename Receiver2>
-        explicit operation(
-            timed_single_thread_context* scheduler,
-            clock_t::time_point dueTime,
-            Receiver2&& receiver)
-            : task_base(scheduler), receiver_((Receiver2 &&) receiver) {
-          this->dueTime_ = dueTime;
-        }
+    explicit type(
+        timed_single_thread_context& context,
+        Duration duration) noexcept
+      : context_(&context), duration_(duration) {}
 
-        void execute() noexcept final {
-          cancelCallback_.destruct();
-          if constexpr (is_stop_never_possible_v<
-                            stop_token_type_t<Receiver&>>) {
-            unifex::set_value(
-                static_cast<Receiver&&>(receiver_), scheduler{this->context_});
-          } else {
-            if (get_stop_token(receiver_).stop_requested()) {
-              unifex::set_done(static_cast<Receiver&&>(receiver_));
-            } else {
-              unifex::set_value(static_cast<Receiver&&>(receiver_));
-            }
-          }
-        }
+    timed_single_thread_context* context_;
+    Duration duration_;
+
+   public:
+    template <
+        template <typename...> class Variant,
+        template <typename...> class Tuple>
+    using value_types = Variant<Tuple<>>;
+
+    template <template <typename...> class Variant>
+    using error_types = Variant<>;
+
+    template <typename Receiver>
+    after_operation<Duration, Receiver> connect(Receiver&& receiver) {
+      return after_operation<Duration, Receiver>{
+          *context_, duration_, (Receiver &&) receiver};
+    }
+  };
 
-        UNIFEX_NO_UNIQUE_ADDRESS Receiver receiver_;
-        UNIFEX_NO_UNIQUE_ADDRESS manual_lifetime<typename stop_token_type_t<
-            Receiver&>::template callback_type<cancel_callback>>
-            cancelCallback_;
-      };
-
-     public:
-      template <typename Receiver>
-      operation<std::remove_cvref_t<Receiver>> connect(Receiver&& receiver) {
-        return operation<std::remove_cvref_t<Receiver>>{
-            context_, dueTime_, (Receiver &&) receiver};
+  template <typename Receiver>
+  struct at_op_ {
+    class type;
+  };
+  template <typename Receiver>
+  using at_operation = typename at_op_<std::remove_cvref_t<Receiver>>::type;
+
+  template <typename Receiver>
+  class at_op_<Receiver>::type final : task_base {
+    template <typename Receiver2>
+    explicit type(
+        timed_single_thread_context& scheduler,
+        clock_t::time_point dueTime,
+        Receiver2&& receiver)
+        : task_base(scheduler)
+        , receiver_((Receiver2 &&) receiver) {
+      this->dueTime_ = dueTime;
+    }
+
+    void execute() noexcept final {
+      cancelCallback_.destruct();
+      if constexpr (is_stop_never_possible_v<
+                        stop_token_type_t<Receiver&>>) {
+        unifex::set_value(
+            static_cast<Receiver&&>(receiver_), scheduler{this->context_});
```

Comment:
This doesn't seem right. The `schedule_at` sender should not be sending a value.
eg. See the call to set_value() in the else-class below.

I suspect that we're never actually instantiating this code-path or otherwise we'd be seeing compile-errors.

Reply:
Shall I fix it here?

Reply:
Fixed